File: src/app/services/category.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseUrl from './helper';

@Injectable({
  providedIn: 'root',
})
export class CategoryService {
  //  HTTP Client Constructor
  constructor(private _http: HttpClient) {}

  //  Load all the categories
  public categories() {
    return this._http.get(`${baseUrl}/category/`);
  }

  //  Add new category
  public addCategory(category: any) {
    return this._http.post(`${baseUrl}/category/`, category);
  }

  //  Delete category
  public deleteCategory(cid: any) {
    return this._http.delete(`${baseUrl}/category/${cid}`);
  }

  //  Get the Single Category
  public getCategory(cid: any) {
    return this._http.get(`${baseUrl}/category/${cid}`);
  }

  //  Update Category
  public updateCategory(category: any) {
    return this._http.put(`${baseUrl}/category/`, category);
  }
}

File: ExamServer/src/main/java/com/exam/service/CategoryService.java
package com.exam.service;

import com.exam.model.exam.Category;

import java.util.Set;

public interface CategoryService {

    //  Add Particular Category
    public Category addCategory(Category category);

    //  Update Particular Category
    public Category updateCategory(Category category);

    //  Get All Categories
    public Set<Category> getCategories();

    //  Get a Particular Category
    public Category getCategory(Long categoryId);

    //  Delete a Particular Category
    public void deleteCategory(Long categoryId);
}
